### DISCUSSION TITLE
Migrating to workerd?

### DISCUSSION THREAD

**Discussion by julienperret** [allmaps/allmaps#65] [Ideas for Data Model, API and Backend]
Is there a plan for migrating to [workerd](https://github.com/cloudflare/workerd) (Cloudflare Open Source Workers runtime)?

It seams it would make it a lot easier to use the tileserver in other hosting environments. At the moment, I find it difficult to use without Cloudflare workers (local or self-host).

**Comment by bertspaan** [comment]
Great idea! I've heard about workerd, but I haven't tried / used it - do you have an idea how much work it'll be to make Allmaps Tile Server work with workerd?

**Comment by julienperret** [comment]
I have no idea but I can try to have a look

**Comment by bertspaan** [comment]
That would be great! I don't have time/funding for significant updates to the tile server, but I do have funding for a new part of Allmaps, very similar to the tile server: a service that can generate PNGs of warped maps. We are planning to use this to create social media previews of maps and other Allmaps URLs. Planning to start working on this service next month, it makes sense to use workerd for this, and then at some point update the tile server to use workerd as well.

**Reply by julienperret** [reply]
I have a working test version here: https://github.com/soduco/allmaps/tree/workerd
It was surprisingly easy so I'm suspecting it might not work.
I'll try to deploy a dev if I have the time. Do you have a recipe I could reproduce on my server?
